### ISSUE TITLE
Make our GCE image pass the GCE requirements for official images

### ISSUE THREAD

**Issue by margamanterola** [flatcar/Flatcar#186]
**Current situation**
This page details the requirements for official GCE images: https://cloud.google.com/marketplace/docs/partners/vm/build-vm-image#test_the_vm_image

From the checks listed there we are missing:
* Having python installed
* Having a working gcloud / gsutil (a docker error is printed when calling these commands)
* Having `dhcp` in the process list. I'm not sure why they want that, though. We use systemd-networkd, maybe that was not considered when creating the list.

**Impact**
We can't provide official images until we pass this validation

**Ideal future situation**
GCE images include python and gsutil/gcloud work out of the box.

/cc @blixtra

**Comment by margamanterola** [comment]
These are the errors printed when trying to execute gsutil or gcloud:

```
marga_kinvolk_io@flatcar-container-linux-1-vm ~ $ gsutil
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.38/images/json?filters=%7B%22reference%22%3A%7B%22google%2Fcloud-sdk%22%3Atrue%7D%7D: dial unix /var/run/docker.sock: connect: permission denied
Warning: failed to get default registry endpoint from daemon (Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.38/info: dial unix /var/run/docker.sock: connect: permission denied). Using system default: https://index.docker.io/v1/
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.38/images/create?fromImage=google%2Fcloud-sdk&tag=latest: dial unix /var/run/docker.sock: connect: permission denied
/run/torcx/bin/docker: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.38/containers/create: dial unix /var/run/docker.sock: connect: permission denied.
See '/run/torcx/bin/docker run --help'.
```

I thought this could be due to the docker service not being running, but starting it didn't make a difference:
```
marga_kinvolk_io@flatcar-container-linux-1-vm ~ $ sudo systemctl start docker
marga_kinvolk_io@flatcar-container-linux-1-vm ~ $ gsutil
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.38/images/json?filters=%7B%22reference%22%3A%7B%22google%2Fcloud-sdk%22%3Atrue%7D%7D: dial unix /var/run/docker.sock: connect: permission denied
Warning: failed to get default registry endpoint from daemon (Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Get http://%2Fvar%2Frun%2Fdocker.sock/v1.38/info: dial unix /var/run/docker.sock: connect: permission denied). Using system default: https://index.docker.io/v1/
Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.38/images/create?fromImage=google%2Fcloud-sdk&tag=latest: dial unix /var/run/docker.sock: connect: permission denied
/run/torcx/bin/docker: Got permission denied while trying to connect to the Docker daemon socket at unix:///var/run/docker.sock: Post http://%2Fvar%2Frun%2Fdocker.sock/v1.38/containers/create: dial unix /var/run/docker.sock: connect: permission denied.
See '/run/torcx/bin/docker run --help'.
```
And running with sudo simply says it can't find the command:

```
marga_kinvolk_io@flatcar-container-linux-1-vm ~ $ sudo gsutil
sudo: gsutil: command not found
```

This is because they are aliases, not real commands:

```
marga_kinvolk_io@flatcar-container-linux-1-vm ~ $ alias -p
alias egrep='egrep --colour=auto'
alias fgrep='fgrep --colour=auto'
alias gcloud='(docker images google/cloud-sdk || docker pull google/cloud-sdk) > /dev/null;docker run -ti --rm --net=host -v /home/marga_kinvolk_io/.config:/root/.config -v /var/run/docker.sock:/var/run/docker.sock google/cloud-sdk gcloud'
alias grep='grep --colour=auto'
alias gsutil='(docker images google/cloud-sdk || docker pull google/cloud-sdk) > /dev/null;docker run -ti --rm --net=host -v /home/marga_kinvolk_io/.config:/root/.config google/cloud-sdk gsutil'
alias ls='ls --color=auto'
```


```

**Comment by margamanterola** [comment]
Summary of what was discussed in slack:

We have two packages in the coreos-overlay repo that handle GCE images: `oem-gce` and `coreos-oem-gce`. The first one is the one used to customize GCE images. The second one is used to create a container that then runs with rkt.

The aliases come from `coreos-base/oem-gce/files/files/google-cloud-sdk.sh`, which is copied to `/etc/profile.d/google-cloud-sdk.sh` by the OEM Ignition file.

**Comment by margamanterola** [comment]
The rkt container that runs with the oem-gce service is intended for the "guest tools" that run on GCE instances, it's not intended to be used by normal users to run commands in it.  Also, it's quite outdated, we probably do want to run a more updated version of it.

As we've discussed with Kai, we could ship an alias that runs python with a docker container. It would be similar to the gcloud/gsutil aliases.

I found why those aliases didn't work for me: I needed to add my user to the docker group. We should try to find a way to do this automatically, so things work out of the box.
